packages/metro-transform-worker/src/__tests__/index-test.js

```diff
@@ -583,3 +584,23 @@ it('counts all line endings correctly', async () => {
     standardEndingsResult.output[0].data.lineCount,
   );
 });
+
+it('allows preservation of comments', async () => {
+  const result = await Transformer.transform(
+    {...baseConfig, unstable_preserveComments: true},
+    '/root',
+    'local/file.js',
+    Buffer.from('/*#__PURE__*/arbitrary(code);', 'utf8'),
+    // $FlowFixMe[prop-missing] Added when annotating Transformer.
+    {
+      dev: false,
+      minify: false,
```

Comment:
We probably need a test for `minify: true` to clarify how these options interact.

Also, 💡 maybe we should just preserve all comments when not minifying? Similar to Babel's defaults: https://babeljs.io/docs/options#shouldprintcomment

Reply:
(By which I mean, maybe we can avoid introducing this option altogether.)

Reply:
I think there's a case for moving this responsibility entirely to the minifier, and just setting `comments: true`. For a case like this when you need comments in the serializer, that shouldn't prevent subsequent minification.

Reply:
I think minify and comments should be two separate things. I'm fine with `minify` affecting the default value of `preserveComments` though. @robhogan's suggestion seems to be the best option here. I can set `comments: true` in `metro-transform-worker`, and `comments: false` in the default minifier config to preserve the current behaviour. Let me know what you prefer.

Reply:
> I think minify and comments should be two separate things.

Can you elaborate? What's the use case?

My preference is for the following setup:
1. The transformer should always preserve comments (before minification) if it can, without us adding a new option to control this.
2. The minifier should default to stripping comments, but this can be changed using the already-existing [`minifierConfig`](https://facebook.github.io/metro/docs/configuration/#minifierconfig) option.

Reply:
I think we are in agreement. I just misunderstood your first comment. 😄

Reply:
Updated. Let me know if I've misunderstood anything.